c7n/filters/offhours.py

```diff
@@ -263,17 +262,21 @@ def __call__(self, i):
             self.enabled_count += 1
 
         try:
-            return self.process_resource_schedule(i, value)
+            return self.process_resource_schedule(i, value, self.time_type)
         except:
             log.exception(
                 "%s failed to process resource:%s value:%s",
                 self.__class__.__name__, i[self.id_key], value)
             return False
 
-    def process_resource_schedule(self, i, value):
+    def process_resource_schedule(self, i, value, time_type):
         """Does the resource tag schedule and policy match the current time."""
         rid = i[self.id_key]
-        if self.parser.has_resource_schedule(value):
+        # this is to normalize trailing semicolons which when done allows
+        # dateutil.parser.parse to process: value='off=(m-f,1);' properly.
+        # before this normalization, some cases would silently fail.
+        value = ';'.join(filter(None, value.split(';')))
```

Comment:
This sanitizing is helpful. If you have a trailing semicolon without this sanitizing, it returns None.. like so:
ipdb> value
'off=(m-f,1);'
ipdb> self.parser.parse(value)
ipdb>
but after the sanitizing.. all the various combinations return the proper values:
ipdb> self.parser.parse(';'.join(filter(None, 'off=(m-f,1);'.split(';'))))
{'tz': 'pt', 'off': [{'days': [0, 1, 2, 3, 4], 'hour': 1}]}
ipdb> self.parser.parse(';'.join(filter(None, 'off=(m-f,1)'.split(';'))))
{'tz': 'pt', 'off': [{'days': [0, 1, 2, 3, 4], 'hour': 1}]}
ipdb> self.parser.parse(';'.join(filter(None, 'off=(m-f,1);on=(m-f,3)'.split(';'))))
{'on': [{'days': [0, 1, 2, 3, 4], 'hour': 3}], 'off': [{'days': [0, 1, 2, 3, 4], 'hour': 1}], 'tz': 'pt'}
ipdb> self.parser.parse(';'.join(filter(None, 'off=(m-f,1);on=(m-f,3);'.split(';'))))
{'on': [{'days': [0, 1, 2, 3, 4], 'hour': 3}], 'off': [{'days': [0, 1, 2, 3, 4], 'hour': 1}], 'tz': 'pt'}
ipdb> self.parser.parse(';'.join(filter(None, 'off=(m-f,1);on=(m-f,3);tz=est'.split(';'))))
{'on': [{'days': [0, 1, 2, 3, 4], 'hour': 3}], 'off': [{'days': [0, 1, 2, 3, 4], 'hour': 1}], 'tz': 'est'}
ipdb> self.parser.parse(';'.join(filter(None, 'off=(m-f,1);on=(m-f,3);tz=est;'.split(';'))))
{'on': [{'days': [0, 1, 2, 3, 4], 'hour': 3}], 'off': [{'days': [0, 1, 2, 3, 4], 'hour': 1}], 'tz': 'est'}

Which means we can only set the value to on or off, and not both, with or without the trailing semicolo, and it works. as well, this is backwards compatible with any existing values for onhours or offhours.

I'm guessing this is why the docs in the function show you need on and off in the value, because if you did just on with a semicolon, it would fail?